src/commands/whoami.rs

```diff
@@ -116,7 +116,24 @@ pub(crate) fn fetch_accounts(user: &GlobalUser) -> Result<Vec<Account>> {
     let response = client.request(&account::ListAccounts { params: None });
     match response {
         Ok(res) => Ok(res.result),
-        Err(e) => anyhow::bail!(http::format_error(e, None)),
+        Err(e) => {
+            match e {
+                ApiFailure::Error(_, ref api_errors) => {
+                    let error = &api_errors.errors[0];
+                    if error.code == 9109 {
+                        // 9109 error code = Invalid access token
+                        StdOut::info("Token might be expired or might not have the necessary permissions. Please consider running `wranger logout` or re-authenticate wrangler through `wrangler login` or `wrangler config`.");
+                    } else if error.code == 6003 {
+                        // 6003 error code = Invalid request headers. A common case is when the value of an authorization method has been changed outside of wrangler commands
+                        StdOut::info("Your authentication method might be corrupted (e.g. API token value has been altered). Please consider re-authenticating wrangler through `wrangler login` or `wrangler config`.")
```

Comment:
Similar suggestion here. Maybe we shouldn't recommend logging out, when they'll have to do a login again anyway (and if login wipes out the previous token)